src/st/np3/4B018.c

```diff
@@ -128,14 +274,134 @@ s32 func_801CE120(Entity* self, s32 arg1) {
     return ret;
 }
 
-INCLUDE_ASM("asm/us/st/np3/nonmatchings/4B018", func_801CE1E8);
+void func_801CE1E8(s16 step) {
+    s32 i;
 
-INCLUDE_ASM("asm/us/st/np3/nonmatchings/4B018", func_801CE228);
+    g_CurrentEntity->step = step;
+    g_CurrentEntity->step_s = 0;
+    g_CurrentEntity->animFrameIdx = 0;
+    g_CurrentEntity->animFrameDuration = 0;
 
-INCLUDE_ASM("asm/us/st/np3/nonmatchings/4B018", func_801CE258);
+    for (i = 0; i < 4; i++) {
+        g_CurrentEntity->ext.GH_Props.unkB0[i] = 0;
+        g_CurrentEntity->ext.GH_Props.unkB0[i + 2] = 0;
+    }
+}
+
+void func_801CE228(s16 step) {
+    s32 i;
+
+    for (i = 0; i < 4; i++) {
+        g_CurrentEntity->ext.GH_Props.unkB0[i] = 0;
```

Comment:
This seems odd to me since the assignment addresses are overlapping, but I guess that may just be what they wrote.

Reply:
That's precisely why i don't want to create entity extensions at decompilation time, because it forces you to make assumptions and can lead in a sense of false documentation. I just checked and both functions are called from Gurkha but only one of them is called from Hammer. Until we have those decompiled we can't make any assumptions on documentation.